BeeSwift/GoalViewController.swift

```diff
@@ -209,30 +209,15 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
         self.dateStepper.tintColor = UIColor.Beeminder.gray
         dataEntryView.addSubview(self.dateStepper)
         self.dateStepper.addTarget(self, action: #selector(GoalViewController.dateStepperValueChanged), for: .valueChanged)
-        self.dateStepper.value = 0
-
-        // if the goal's deadline is after midnight, and it's after midnight,
-        // but before the deadline,
-        // default to entering data for the "previous" day.
-        let now = Date()
-        let calendar = Calendar.current
-        let components = (calendar as NSCalendar).components([.hour, .minute], from: now)
-        let currentHour = components.hour
-        if self.goal.deadline > 0 && currentHour! < 6 && currentHour! < self.goal.deadline/3600 {
-            self.dateStepper.value = -1
-        }
-
-        // if the goal's deadline is before midnight and has already passed for this calendar day, default to entering data for the "next" day
-        if self.goal.deadline < 0 {
-            let deadlineSecondsAfterMidnight = 24*3600 + self.goal.deadline
-            let deadlineHour = deadlineSecondsAfterMidnight/3600
-            let deadlineMinute = (deadlineSecondsAfterMidnight % 3600)/60
-            let currentMinute = components.minute
-            if deadlineHour < currentHour! ||
-                (deadlineHour == currentHour! && deadlineMinute < currentMinute!) {
-                self.dateStepper.value = 1
-            }
-        }
+        self.dateStepper.value = {
```

Comment:
I agree this is an improvement and seems reasonable to merge. Long term we should probably change how the dateStepper works to use a representation more tied to daystamps and less to the current clock time.